packages/nx/src/plugins/js/project-graph/build-dependencies/explicit-package-json-dependencies.ts

```diff
@@ -80,29 +109,80 @@ function processPackageJson(
         };
         validateDependency(dependency, ctx);
         collectedDeps.push(dependency);
-      } else if (ctx.externalNodes[`npm:${d}`]) {
-        const dependency: RawProjectGraphDependency = {
-          source: sourceProject,
-          target: `npm:${d}`,
-          sourceFile: fileName,
-          type: DependencyType.static,
-        };
-        validateDependency(dependency, ctx);
-        collectedDeps.push(dependency);
+        continue;
       }
-    });
+
+      try {
+        // package.json refers to an external package, we do not match against the version found in there, we instead try and resolve the relevant package how node would
+        let externalPackageVersion: string | undefined;
+        const externalPackageJsonPath = findExternalPackageJsonPath(
+          d,
+          dirname(fileName)
+        );
+        // The package.json path might be not be resolvable, e.g. if a reference has been added to the package.json, but the install command has not been run yet.
+        if (!externalPackageJsonPath) {
+          continue;
+        }
+        if (!externalPackageJsonCache.has(externalPackageJsonPath)) {
+          const externalPackageJson = parseJson(
+            readFileSync(externalPackageJsonPath, 'utf-8')
+          );
+          externalPackageJsonCache.set(
+            externalPackageJsonPath,
+            externalPackageJson
+          );
+          externalPackageVersion = externalPackageJson.version;
+        } else {
+          externalPackageVersion = externalPackageJsonCache.get(
```

Comment:
Hm... could we keep the cache management and usage in the module scope of the function rather than passing it around?

Reply:
- I didn't want to run any risks of stale caches via require caching
- Unit tests are more straightforward/direct